TourMate/TourMate/Frontend/Plan/Views/PlanViews/PlanViewModel.swift

```diff
@@ -98,6 +98,31 @@ class PlanViewModel<T: Plan>: PlanDisplayViewModel<T> {
         await update(plan: plan, errorMessage: "")
     }
 
+    func restoreToCurrentVersion() async {
+        guard let latestVersion = allVersionNumbers.max() else {
+            handleError()
+            return
+        }
+
+        let (currentUser, userError) = await userService.getCurrentUser()
+
+        guard let currentUser = currentUser, userError.isEmpty else {
+            handleError()
+            return
+        }
+
+        plan.versionNumber = latestVersion + 1
+        plan.creationDate = Date()
+        plan.modificationDate = Date()
+        plan.modifierUserId = currentUser.id
```

Comment:
Would it be better to create a new plan instead of modifying the existing one? Because for some reason if the addition fails, the plan that is being displayed will be not in the database

Reply:
I think its fine, I dismiss the view after restoring version.

Reply:
Okay sure then no issue there I guess

Reply:
Oh yeah no need to update creation date